### PR TITLE
espeak-ng: Upgrade package version to 1.51.1 

### PR DESCRIPTION
**Pull request by Sumynwa**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- x ] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- x ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 
upgrade package version to 1.51.1

NOTE: check section fails for this version due to known issue: https://github.com/espeak-ng/espeak-ng/issues/1270

###### Change Log  
- bump package version
- remove unneeded patch files
- lint spec
- update cgmanifest entry

###### Does this affect the toolchain?  
NO

###### Test Methodology

- Pipeline build id: BUDDY BUILD
  https://dev.azure.com/mariner-org/mariner/_build/results?buildId=494900

### CODE REVIEW

File: SPECS/espeak-ng/espeak-ng.spec [hunk 1]
```diff
@@ -1,16 +1,17 @@
 Summary:        Compact text-to-speech synthesizer
 Name:           espeak-ng
-Version:        1.50
-Release:        3%{?dist}
+Version:        1.51.1
+Release:        1%{?dist}
 # Apache2 license applies only to Android APK code- does not apply here
 # BSD license applies only to Windows code- does not apply here
-License:        GPLv3 AND Unicode
+License:        GPL-3.0-only AND GPL-3.0-or-later AND Unicode-DFS-2016 AND CC-BY-SA-3.0
```
Comment (ameyapu) [inline SPECS/espeak-ng/espeak-ng.spec:7]: I didn't get this change.  Are you saying GPLv3 representation has changed?

File: SPECS/espeak-ng/espeak-ng.spec [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 URL:            https://github.com/espeak-ng/espeak-ng
-Source0:        https://github.com/%{name}/%{name}/releases/download/%{version}/%{name}-%{version}.tgz
-Patch0:         tests-fix-greek-letter-variants.patch
-Patch1:         tests-newline-fixes.patch
+Source0:        https://github.com/%{name}/%{name}/archive/%{version}/%{name}-%{version}.tar.gz#/%{name}-%{version}.tgz
+# Backported from:
+# https://github.com/espeak-ng/espeak-ng/commit/58f1e0b6a4e6aa55621c6f01118994d01fd6f68c
```
Comment (ameyapu) [inline SPECS/espeak-ng/espeak-ng.spec:13]: If these are not needed, we can remove these lines.

**Review by ameyapu — APPROVED** [review]
LGTM.

File: SPECS/espeak-ng/espeak-ng.spec (hunk 2 above)
Comment (Sumynwa) [reply SPECS/espeak-ng/espeak-ng.spec:13]: Thanks. Not needed will remove this.

File: SPECS/espeak-ng/espeak-ng.spec (hunk 1 above)
Comment (Sumynwa) [reply SPECS/espeak-ng/espeak-ng.spec:7]: No, I will update this.

**Pull request merged**